haproxy/state/from_ha.go

```diff
@@ -51,30 +51,44 @@ func FromHAProxy(ha HAProxyRead) (State, error) {
 		if err != nil {
 			return state, err
 		}
-		if len(logTargets) > 1 {
-			return state, fmt.Errorf("expected at most 1 filter for frontend %s, got %d", f.Name, len(filters))
+		if len(logTargets) >= 2 {
+			return state, fmt.Errorf("expected at most 2 filters for frontend %s, got %d", f.Name, len(filters))
 		}
-		var filter *FrontendFilter
-		if len(filters) == 1 {
-			filter = &FrontendFilter{
-				Filter: filters[0],
+		var filter_spoe *FrontendFilter
+		var filter_compression *FrontendFilter
+		if len(filters) <= 2 {
+			index_compression_filter := 0
+			index_spoe_filter := 1
+			if filters[0].Type == models.FilterTypeSpoe {
```

Comment:
What's the goal of this block? Could you please add a comment to explain this inversion?

Reply:
Simplified